homeassistant/components/osoenergy/config_flow.py

```diff
@@ -72,6 +61,9 @@ async def async_step_reauth(
         self, entry_data: Mapping[str, Any]
     ) -> ConfigFlowResult:
         """Re Authenticate a user."""
-        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
-        data = {CONF_API_KEY: entry_data[CONF_API_KEY]}
-        return await self.async_step_user(data)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                _SCHEMA_STEP_USER, self._get_reauth_entry().data
+            ),
+        )
```

Comment:
Small change in behavior here: I've removed the auto-attempt of the old API key.
I think the user should have the ability to confirm before trying the API key again